app/lib/katello/validators/content_view_puppet_module_validator.rb

```diff
@@ -3,21 +3,26 @@ module Validators
     class ContentViewPuppetModuleValidator < ActiveModel::Validator
       def validate(record)
         if record.uuid.blank? && (record.name.blank? || record.author.blank?)
-          invalid_parameters = _("Invalid puppet module parameters specified.  Either 'uuid' or 'name' and 'author' must be specified.")
-          record.errors[:base] << invalid_parameters
-          return
-        end
-
-        if record.name && record.author
-          # validate that a puppet module exists with this name+author
-          unless PuppetModule
-                  .in_repositories(record.content_view.puppet_repos)
-                  .where(:name => record.name, :author => record.author).present?
-
-            invalid_parameters = _("Puppet Module with name='%{name}' and author='%{author}' does not exist") %
-                { :name => record.name, :author => record.author }
+          record.errors[:base] << _("Invalid puppet module parameters specified. \
+                                    Either 'uuid' or 'name' and 'author' must be specified.")
+        elsif record.name && record.author &&
+          !PuppetModule.exists?(name: record.name, author: record.author)
+          record.errors[:base] << _("Puppet Module with name='%{name}' and author='%{author}' does\
+                                    not exist") % { name: record.name, author: record.author }
+        elsif record.uuid && !PuppetModule.exists?(uuid: record.uuid)
+          record.errors[:base] << _("Puppet Module with uuid='%{uuid}' does not\
+                                    exist") % { uuid: record.uuid }
+        else
+          repositories = PuppetModule.where(
+            record.uuid.blank? ?
+            { name: record.name, author: record.author } :
+            { uuid: record.uuid }
+          ).flat_map(&:repositories)
```

Comment:
How about this, @ehelms? I'm trying to avoid having to say `repositories = repositories.flat_map(&:repositories)` because that also is confusing. I'd have to use another variable called `puppet_modules`. Anyway, I can iterate again if you think this is also confusing or hard to read.